### PR TITLE
Added ObjectShapeType

### PR DESCRIPTION
**Pull request by vudaltsov**

### CODE REVIEW

File: src/types.php [hunk 1]
```diff
@@ -308,6 +308,28 @@ public static function iterable(Type $keyType = self::mixed, Type $valueType = s
         return new IterableType($keyType, $valueType);
     }
 
+    /**
+     * @psalm-pure
+     * @param array<string, Property> $properties
```
Comment (klimick) [inline src/types.php]: Should be `@param array<string, Type|Property> $properties`

Comment (vudaltsov) [reply src/types.php]: Thanks, fixed!

File: src/types.php [hunk 2]
```diff
@@ -308,6 +308,28 @@ public static function iterable(Type $keyType = self::mixed, Type $valueType = s
...
+                static fn (Type|Property $property): Property => $property instanceof Type ? new Property($property) : $property,
+                $properties,
+            ),
+        );
+    }
+
+    /**
+     * @psalm-pure
+     */
+    public static function prop(Type $type, bool $optional = false): Property
```
Comment (vudaltsov) [inline src/types.php:328]: Should we use full names here?..

Comment (klimick) [reply src/types.php:328]: I can't say why, but I like the full name.
Maybe `prop` sounds like `a` or `b` for variable name.

Comment (vudaltsov) [reply src/types.php:328]: Okay, agree. I will merge it here and then change all short names in master.

**Pull request merged**